aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -135,6 +138,13 @@ void multi_tensor_apply(
   using scalar_vals_t = typename T::opmath_t;
   TensorListScalarListMetadata<scalar_vals_t, depth> tensorListMeta;
 
+  const auto needs_64bits_for_indexing = std::any_of(
+      tensor_lists[0].cbegin(),
+      tensor_lists[0].cend(),
+      [](const auto& t) -> bool {
+        return t.numel() > std::numeric_limits<int32_t>::max();
```

Comment:
should this be?
```suggestion
        return t.numel() > std::numeric_limits<uint32_t>::max();
```
Not sure whether we use a signed or unsigned 32bit int for indexing.

Reply:
I don't remember if there is a rule about it but I switched it unsigned int's